app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+        self, name: str, health: int = 100, hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            "{Name: "
+            + self.name
+            + ", Health: "
+            + str(self.health)
+            + ", Hidden: "
+            + str(self.hidden)
+            + "}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden is False:
+            self.hidden = True
+        elif self.hidden is True:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+    def bite(self, victim: Animal) -> None:
+        if victim.hidden is False and isinstance(victim, Herbivore):
```

Comment:
u don't need to use id comparison in this case
```suggestion
        if not victim.hidden and isinstance(victim, Herbivore):
```